src/main/java/org/rcsb/cif/CifIO.java

```diff
@@ -128,7 +129,8 @@ public static CifFile readFromInputStream(InputStream inputStream) throws IOExce
      */
     public static CifFile readFromInputStream(InputStream inputStream, CifOptions options) throws IOException {
         // performance: explicitly buffer stream, increases performance drastically
-        if (!(inputStream instanceof BufferedInputStream)) {
+        if (!(inputStream instanceof BufferedInputStream)
+        		&& !(inputStream instanceof ByteArrayInputStream)) {
```

Comment:
Good call that a `ByteArrayInputStream` doesn't need the `BufferedInputStream` decorator.